df_udf/README.md

```diff
@@ -0,0 +1,90 @@
+# Scala / Java UDFS implemented using data frame
+
+User Defined Functions (UDFs) are used for a number of reasons in Apache Spark. Much of the time it is to implement 
+logic that is either very difficult or impossible to implement using existing SQL/Dataframe APIs directly. But they
+are also used as a way to standardize processing logic across an organization or for code reused.
+
+But UDFs come with some downsides. The biggest one is visibility into the processing being done. SQL language that
+can be highly optimized. But a UDF in most cases is a black box, that the SQL optimizer cannot do anything about.
+This can result in less than ideal query planning. Additionally, accelerated execution environments, like the
+RAPIDS Accelerator for Apache Spark have not easy way to replace UDFs with accelerated versions, which can result in
+slow performance.
+
+This attempts to add visibility to the code reuse use case by providing a way to implement a UDF in terms of dataframe
+commands. 
+
+## Setup
+
+To do this include com.nvidia:df_udf_plugin as a dependency for your project and also include it on the 
+classpath for your Apache Spark environment. Then include `com.nvidia.spark.DFUDFPlugin` in the config 
+`spark.sql.extensions`. Now you can implement a UDF in terms of Dataframe operations.
```

Comment:
Is there going to be a separate jar for this, or is it always bundled with the RAPIDS Accelerator jar?  If it's always going to run with the RAPIDS Accelerator, wondering if we can avoid needing the user to setup spark.sql.extensions for this and instead set a RAPIDS config.  I can see some utility in using this outside of the RAPIDS Accelerator, but that implies a separate jar not mentioned here.